src/rgw/rgw_perf_counters.h

```diff
@@ -58,3 +59,54 @@ enum {
   l_rgw_last,
 };
 
+enum {
+  l_rgw_op_first = 16000,
+
+  l_rgw_op_put,
+  l_rgw_op_put_b,
+  l_rgw_op_put_lat,
+
+  l_rgw_op_get,
+  l_rgw_op_get_b,
+  l_rgw_op_get_lat,
+
+  l_rgw_op_del_obj,
+  l_rgw_op_del_obj_b,
+  l_rgw_op_del_obj_lat,
+
+  l_rgw_op_del_bucket,
+  l_rgw_op_del_bucket_lat,
+
+  l_rgw_op_copy_obj,
+  l_rgw_op_copy_obj_b,
+  l_rgw_op_copy_obj_lat,
+
+  l_rgw_op_list_obj,
+  l_rgw_op_list_obj_lat,
+
+  l_rgw_op_list_buckets,
+  l_rgw_op_list_buckets_lat,
+
+  l_rgw_op_last
+};
+
+namespace rgw::op_counters {
+
+struct CountersContainer {
+  std::shared_ptr<PerfCounters> user_counters;
+  std::shared_ptr<PerfCounters> bucket_counters;
+};
+
+extern PerfCounters *global_op_counters;
+
+void global_op_counters_init(CephContext *cct);
+
+CountersContainer get(req_state *s);
+
+void inc(CountersContainer user_bucket_counters, int idx, uint64_t v);
+
+void tinc(CountersContainer user_bucket_counters, int idx, utime_t);
+
+void tinc(CountersContainer user_bucket_counters, int idx, ceph::timespan amt);
```

Comment:
shared_ptrs are expensive to copy because they require an atomic ref/deref. taking `CountersContainer` by-value in these functions means two shared_ptr copies each time. take them by const-ref instead